### PR TITLE
Added placeholder label

### PR DESCRIPTION
**Pull request by KrakenZ**
```lua
DTextEntry:SetPlaceholderText( strValue )
```

Placeholder text won't show if the DTextEntry has text.
example

### CODE REVIEW

**Comment by meharryp** [comment]
I think it should be a bit darker like this: http://i.imgur.com/DbPnamr.png
But otherwise, +1

**Comment by Donkie** [comment]
You should use the C-style operators. && instead of and, and || instead of or

Also you should probably call surface.SetFont() before running the surface.GetTextSize

**Comment by KrakenZ** [comment]
Oh yeah right

**Comment by Donkie** [comment]
Just to notify, Robotboy doesn't accept pull requests unless they match the existing code styling.

**Comment by DrVrej** [comment]
Why is this still not a thing?

File: garrysmod/lua/vgui/dtextentry.lua [hunk 1]
```diff
@@ -47,6 +48,23 @@ function PANEL:Init()
 	self.History = {}
 	self.HistoryPos = 0
 	
+	self.PlaceholderText = vgui.Create( "DLabel", self )
+	self.PlaceholderText:Dock( FILL )
+	self.PlaceholderText:DockMargin( 3, 1, 3, 0 )
+	self.PlaceholderText:SetMouseInputEnabled( false )
+	self.PlaceholderText:SetText( "" )
+	self.PlaceholderText:SetWrap( true )
+	self.PlaceholderText:SetTextColor( Color( 169, 169, 169 ) )
```
Comment (Bo98) [inline garrysmod/lua/vgui/dtextentry.lua]: This should be an configurable. Preferably through both Derma skins (see `SKIN:SchemeTextEntry`) and as a PANEL AccessorFunc.

File: garrysmod/lua/vgui/dtextentry.lua [hunk 2]
```diff
@@ -47,6 +48,23 @@ function PANEL:Init()
...
+	self.PlaceholderText:DockMargin( 3, 1, 3, 0 )
+	self.PlaceholderText:SetMouseInputEnabled( false )
+	self.PlaceholderText:SetText( "" )
+	self.PlaceholderText:SetWrap( true )
+	self.PlaceholderText:SetTextColor( Color( 169, 169, 169 ) )
+	self.PlaceholderText:SetContentAlignment( 4 )
+	self.PlaceholderText.PerformLayout = function( self )
+		
+		self:SetContentAlignment( self:GetParent():IsMultiline() and 7 or 4 )
+		self.m_colText.a = self:GetParent():GetText() == "" and 255 or 0
```
Comment (Bo98) [inline garrysmod/lua/vgui/dtextentry.lua]: Does `self:SetVisible(self:GetParent():GetText() == "")` not work?

File: garrysmod/lua/vgui/dtextentry.lua [hunk 3]
```diff
@@ -310,6 +329,12 @@ function PANEL:SetValue( strValue )
 
 end
 
+function PANEL:SetPlaceholderText( strValue )
+	
+	self.m_sPlaceholderText = tostring( strValue )
```
Comment (Bo98) [inline garrysmod/lua/vgui/dtextentry.lua]: Is that necessary? Why should it have a different behaviour to `Panel:SetText(...)` and `DTextEntry:SetValue`?

**Comment by robotboy655** [comment]
This Pull Request is also no longer can be merged.

File: garrysmod/lua/vgui/dtextentry.lua (hunk 3 above)
Comment (KrakenZ) [inline garrysmod/lua/vgui/dtextentry.lua]: Should we really debate over this small thing?

**Comment by KrakenZ** [comment]
what are the conflicts

**Comment by BlackCetha** [comment]
https://github.com/garrynewman/garrysmod/commits/master/garrysmod/lua/vgui/dtextentry.lua

**Comment by zerfgog** [comment]
@KrakenZ you can either rebase or make a new branch from upstream/master and commit anew

**Comment by Bo98** [comment]
You can fix the conflicts properly by doing a git rebase onto the latest commit in this repo.

**Comment by Bo98** [comment]
As for skins, adding `SKIN.colTextEntryTextPlaceholder = Color(169, 169, 169)` to the skin and overriding UpdateColours for the placeholder label should be enough.

```lua
function self.PlaceholderText:UpdateColours(skin)
    self:SetTextStyleColor(skin.colTextEntryTextPlaceholder)
end
```

**Comment by robotboy655** [comment]
Are you going to do the rebase thing & fix the pointed out stuff or should I close this and redo it myself?

**Comment by zerfgog** [comment]
@robotboy655 You could probably add his fork as a remote then merge his branch into master, then create the usual commit to fix merge conflicts.

**Comment by KrakenZ** [comment]
Sorry but I don't have time for anything gmod-related at this moment. I'll close this pull request.

**Comment by MrUnknownGamer** [comment]
any updates on this?

**Pull request closed without merging**